bluetooth_mesh/apps/meshcli.py

```diff
@@ -958,15 +959,32 @@ async def __call__(self, application, arguments):
 
         model: LightLightnessClient = self.get_model(application)
 
-        await asyncio.gather(
-            model.set_lightness_range_unack(
-                app_index=0,
-                destination=address,
-                min_lightness=min_lightness,
-                max_lightness=max_lightness
+        addresses = self.get_addresses(application, arguments)
+        if not arguments["--ack"]:
+            await asyncio.gather(
+                model.set_lightness_range_unack(
+                    app_index=0,
+                    destination=address,
+                    min_lightness=min_lightness,
+                    max_lightness=max_lightness
+                )
+                for address in addresses
             )
-            for address in self.get_addresses(application, arguments)
-        )
+        else:
+            for address in addresses:
+                try:
+                    result = await model.set_lightness_range(
+                        app_index=0,
+                        destination=address,
+                        min_lightness=min_lightness,
+                        max_lightness=max_lightness
+                    )
+                except asyncio.TimeoutError:
```

Comment:
Also, if you look closely, `Model.query` does something like this.

Reply:
```
        async def sender():
            while not status.done():
                await request()
                await asyncio.sleep(send_interval)

        sender = asyncio.ensure_future(sender())

        try:
            await asyncio.wait_for(status, timeout=timeout)
        finally:
            with suppress(asyncio.CancelledError):
                sender.cancel()
                await sender

        return status.result()
```
The bit I'm missing is where does `status.result()` get set?

Reply:
If I understand the `app_message_received` in `expect_app` correctly, then if the address does not respond, then `return status.result()` throws `InvalidStateError`, because nobody has set its result. NB: `If the Future’s result isn’t yet available, this method raises a InvalidStateError exception.` (https://docs.python.org/3.7/library/asyncio-future.html)

Reply:
I agree that there's no need to catch `TimeoutError`s in app code, as it's already serviced by `Model.query`. It needs to catch `InvalidStateError` instead, is that right?